### DISCUSSION TITLE
Digits in „wrong“ order

### DISCUSSION THREAD

**Discussion by ede-kowalski** [jomjol/AI-on-the-edge-device#2250] [General]
Hi,

maybe i am to stupid but i can do whatever the watermeter recognizes the correct numbers but in the wrong order.
 My understanding is that it should be digit1 from left to digit3 which is the last digit right as i have 3 in total.
Nevertheless the calcilated value is 253 instead of 352. 
I even tried with chaning the digit numbers without any success.  Also changing the models didn‘t help.
Has anyone a hint?
Thanks,
Ede

**Comment by caco3** [comment]
check the order of the ROIs in the config, it should be something like this:
```
[Digits]
...
main.dig1 300 42 58 79 false
main.dig2 358 42 58 79 false
main.dig3 416 42 58 79 false
```

**Comment by ede-kowalski** [comment]
Its:

`[Digits] `
` Model = /config/dig-class100-0150_s2_q.tflite `
` CNNGoodThreshold = 0.5 `
`;LogImageLocation = /log/digit `
` ;LogfileRetentionInDays = 3 `
` main.dig1 355 84 50 77 0 `
` main.dig2 312 84 50 77 0 `
` main.dig3 267 84 50 77 0 `

**Reply by caco3** [reply]
dig1 must be on the left and dig3 on the right:

![grafik](https://user-images.githubusercontent.com/1783586/227806225-df693020-8206-4245-9f85-74750fcd2e40.png)

**Comment by ede-kowalski** [comment]
same for me:
![image](https://user-images.githubusercontent.com/20172640/227902605-4fa8d815-9706-4c85-bebf-d7dae330456e.png)
but under rec:
![image](https://user-images.githubusercontent.com/20172640/227902760-988b4ea6-2367-43d5-aa84-5a7ce083b7e5.png)

**Reply by caco3** [reply]
yes, but the firmware does not care about the naming bot the order in the config file.

I guess the order is different for you in the ROI window, right?
![grafik](https://user-images.githubusercontent.com/1783586/227912375-99994377-987b-4667-aa94-3220eef1da57.png)

Use the move buttons there to reorder or do it manually in the config file.

BTW, your ROIs are not setup correctly!
https://user-images.githubusercontent.com/20172640/227902760-988b4ea6-2367-43d5-aa84-5a7ce083b7e5.png
Please do it according to my screenshot resp. https://jomjol.github.io/AI-on-the-edge-device-docs/ROI-Configuration/#how-to-setup-the-digit-rois-perfectly
